### PR TITLE
RFC: nixos/home-assistant: NixOS is an unsupported installation method

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Trying to steer NixOS users away from reporting bugs to the upstream,
when they don't have the capacity to support bugs that could be the
result of our downstreaming setup.

Probably not the perfect place to put this, but I couldn't think of a
better one yet.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabaff — APPROVED** [review]

**Comment by fabaff** [comment]
In the meantime it's also mentioned in the wiki: https://nixos.wiki/wiki/Home_Assistant

**Comment by mweinelt** [comment]
Also added a section concerning how to run the (downstream-supported) Home Assistant version from NixOS unstable. https://nixos.wiki/wiki/Home_Assistant#Running_a_recent_version_using_an_overlay

And squashed a bug in our dependency evaluation that would make it not consider overwritten package versions.

File: pkgs/servers/home-assistant/default.nix [hunk 1]
```diff
@@ -799,6 +799,7 @@ in with py.pkgs; buildPythonApplication rec {
     tests = {
       inherit (nixosTests) home-assistant;
     };
+    pythonPackages = py.pkgs;
```
Comment (dotlambda) [inline pkgs/servers/home-assistant/default.nix:802]:
```suggestion
    python = py;
```

File: pkgs/servers/home-assistant/parse-requirements.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 from rich.table import Table
 
 COMPONENT_PREFIX = "homeassistant.components"
-PKG_SET = "python3Packages"
+PKG_SET = "home-assistant.pythonPackages"
```
Comment (dotlambda) [inline pkgs/servers/home-assistant/parse-requirements.py:35]:
```suggestion
PKG_SET = "home-assistant.python.pkgs"
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/servers/home-assistant/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix:802]: What would be the benefit of exposing all of it?

Comment (dotlambda) [reply pkgs/servers/home-assistant/default.nix]: I guess it's personal preference but we override `python`, not `pythonPackages`, so I think that should be the thing in `passthru`.

Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix]: WFM.

**Comment by dotlambda** [comment]
Now the commit message needs to be corrected though.

**Comment by mweinelt** [comment]
> Now the commit message needs to be corrected though.

Updated, thanks.

**Comment by dotlambda** [comment]
Some logic in `parse-requirements.py` must be wrong if this actually changes the number of supported components.

**Pull request merged**